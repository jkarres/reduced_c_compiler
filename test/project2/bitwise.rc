// testing |, ^, and &

function : void main() {
  int a = 10;
  int b = 9;
  int c = 8;

  cout << 1 << " " << a | a << endl;
  cout << 2 << " " << a | b << endl;
  cout << 3 << " " << a | c << endl;
  cout << 4 << " " << b | a << endl;
  cout << 5 << " " << b | b << endl;
  cout << 6 << " " << b | c << endl;
  cout << 7 << " " << c | a << endl;
  cout << 8 << " " << c | b << endl;
  cout << 9 << " " << c | c << endl;
  cout << 10 << " " << a ^ a << endl;
  cout << 11 << " " << a ^ b << endl;
  cout << 12 << " " << a ^ c << endl;
  cout << 13 << " " << b ^ a << endl;
  cout << 14 << " " << b ^ b << endl;
  cout << 15 << " " << b ^ c << endl;
  cout << 16 << " " << c ^ a << endl;
  cout << 17 << " " << c ^ b << endl;
  cout << 18 << " " << c ^ c << endl;
  cout << 19 << " " << a & a << endl;
  cout << 20 << " " << a & b << endl;
  cout << 21 << " " << a & c << endl;
  cout << 22 << " " << b & a << endl;
  cout << 23 << " " << b & b << endl;
  cout << 24 << " " << b & c << endl;
  cout << 25 << " " << c & a << endl;
  cout << 26 << " " << c & b << endl;
  cout << 27 << " " << c & c << endl;
}

