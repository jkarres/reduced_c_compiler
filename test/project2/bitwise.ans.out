1 10
2 11
3 10
4 11
5 9
6 9
7 10
8 9
9 8
10 0
11 3
12 2
13 3
14 0
15 1
16 2
17 1
18 0
19 10
20 8
21 8
22 8
23 9
24 8
25 8
26 8
27 8
